### PR TITLE
mpc85xx: handle MSM460 wifi mac in dts

### PR DESCRIPTION
**Pull request by neheb**
Userspace handling is deprecated.

ping @blocktrron

### CODE REVIEW

**Comment by neheb** [comment]
ping @apfelwurm @RaylynnKnight @alexsorinpop @tmn505

**Comment by neheb** [comment]
@JoneKone does this work for you?

**Comment by blocktrron** [comment]
I'll check in coming weeks. This is by no means urgent to have it done in days.

**Comment by neheb** [comment]
Sure. Unlike the other PR, Ethernet driver is not involved.

**Comment by alexsorinpop** [comment]
I think we need to better understand the bid before we do this, since on MSM560 (at least) I found 2 version of BID (one having command line option just "ro" the other one "ro console=ttyS0,115200" this would change the location from where the mac address is read (before "ro" (at location 0x1F87C) there is the length of the actual command line ("ro"=02, "ro console=ttyS0,115200"=17)) and this will change the location of the mac from 0x1f9bd (0x1f9bb + 0x02) to 0x1f9d2 (0x1f9bb + 17).

File: target/linux/mpc85xx/files/arch/powerpc/boot/dts/msm460.dts [hunk 1]
```diff
@@ -190,6 +197,13 @@
 				  0x1000000 0x0 0x0
 				  0x1000000 0x0 0x0
 				  0x0 0x100000>;
+
+			wifi@0,0 {
+				reg = <0x0000 0 0 0 0>;
+				nvmem-cells = <&macaddr_hwinfo_1f9bd 16>;
+				nvmem-cell-names = "mac-address";
+				qca,led-pin = /bits/ 8 <3>;
```
Comment (blocktrron) [inline target/linux/mpc85xx/files/arch/powerpc/boot/dts/msm460.dts:205]: Did you test this on your board? The radio is on a socketed card and although i know you have pins for RF state on the pcie pinout, i don't understand the benefit of this move.

**Review by blocktrron — CHANGES_REQUESTED** [review]

File: target/linux/mpc85xx/files/arch/powerpc/boot/dts/msm460.dts (hunk 1 above)
Comment (alexsorinpop) [reply target/linux/mpc85xx/files/arch/powerpc/boot/dts/msm460.dts:205]: I did test the old version (userspace) and it's not working with 0x1f9bd (on one device). Below is a bid parser that I put together in 10 mins. ID 40 is the MAC address

[bid_parser.py.txt](https://github.com/user-attachments/files/17085088/bid_parser.py.txt)

Comment (neheb) [reply target/linux/mpc85xx/files/arch/powerpc/boot/dts/msm460.dts:205]: The benefit is that now the ath9k LEDs get created with a proper GPIO number. They're created anyway, so might as well create them properly.

Btw, the wdr4900v1 (also mpc85xx) is using this with no problems. I used qca,led-pin there as that's what the platform code was doing before I migrated it to dts.

Comment (blocktrron) [reply target/linux/mpc85xx/files/arch/powerpc/boot/dts/msm460.dts:205]: @neheb The WDR4900 might have the sink wired up to the GPIO and driving it from the ath9k chip. At least that's how i've seen it done sometimes. This is not platform-specific but depends on the electrical design

Comment (neheb) [reply target/linux/mpc85xx/files/arch/powerpc/boot/dts/msm460.dts:205]: right. This needs testing. My suspicion is that it works.

Comment (neheb) [reply target/linux/mpc85xx/files/arch/powerpc/boot/dts/msm460.dts]: Great. Removed wifi change.

**Comment by neheb** [comment]
@blocktrron @robimarko this has been tested as working.

**Comment by blocktrron** [comment]
Merged, thanks!

**Pull request closed without merging**